packages/ui/core-components/src/lib/atoms/inputs/slider/_Slider.svelte

```diff
@@ -0,0 +1,175 @@
+	export let name;
+
+	/** @type {number|undefined} */
+	export let min = 0;
+
+	if (min) {
+		// if min was user-supplied
+		min = Number(min);
+		if (isNaN(min)) {
+			// input must be a number
+			throw Error('min must be a number');
+		}
+	} else {
+		min = 0;
+	}
+
+	/** @type {number} */
+	export let max = 100;
+
+	if (max) {
+		// if min was user-supplied
+		max = Number(max);
+		if (isNaN(max)) {
+			// input must be a number
+			throw Error('max must be a number');
+		} else if (max < min) {
+			throw Error('max cannot be less than min');
+		}
+	} else {
+		max = 100;
+	}
+
+	/** @type {number} */
+	export let step;
+
+	/** @type {boolean} */
+	export let showMaxMin = true;
+	$: showMaxMin = showMaxMin === 'true' || showMaxMin === true;
+
+	/** @type {boolean} */
+	export let hideDuringPrint = true;
+	$: hideDuringPrint = hideDuringPrint === 'true' || hideDuringPrint === true;
+
+	/** @type {number | string} */
+	export let defaultValue = 0;
+
+	/** @type {[number]} */
+	let value = [0];
+
+	if (typeof defaultValue === 'number') {
+		value = [defaultValue];
+	} else if (toNumber(defaultValue)) {
+		value = [toNumber(defaultValue)];
+	}
+
+	/** @type {string | undefined} */
+	export let fmt = undefined;
+
+	$: $inputs[name] = value;
+
+	/** @type {string} */
+	export let size = '';
+
+	const renderSize = (size) => {
+		const sizeMap = {
+			medium: 'w-64',
+			large: 'w-96',
+			//Full size width requires calc to compensate for shifted range span in sliderShadcn
+			full: 'w-[calc(100%-0.6rem)]'
+		};
+		return sizeMap[size.toLowerCase()] || 'w-40';
+	};
+
+	$: sizeClass = renderSize(size);
+
+	let format_object;
+	if (fmt) format_object = getFormatObjectFromString(fmt, 'number');
+	else format_object = undefined;
+
+	// Query data integration
+	export let data = null;
+
+	let error;
+
+	let strictBuild;
+
+	export let maxColumn = undefined;
+	export let minColumn = undefined;
+
+	let initialized = false;
+
+	if (!initialized) {
+		try {
+			error = undefined;
+			if (data) {
+				if (typeof data == 'string') {
+					throw Error(`Received: data=${data}, expected: data={${data}}`);
+				}
+
+				if (!Array.isArray(data)) {
+					// Accept bare objects
+					data = [data];
+				}
+
+				if (maxColumn && data[0]?.[maxColumn]) {
+					max = data[0][maxColumn];
+				}
+
+				if (minColumn && data[0]?.[minColumn]) {
+					min = data[0][minColumn];
+				}
+
+				// Only set value on initial mount
+				if (typeof defaultValue === 'string' && data[0]?.[defaultValue]) {
+					value = [data[0][defaultValue]];
+				}
+				initialized = true;
+			} else if (maxColumn || minColumn) {
+				throw Error(
+					'No data provided. If you referenced a query result, check that the name is correct.'
+				);
+			}
+		} catch (e) {
+			error = e.message;
+			const setTextRed = '\x1b[31m%s\x1b[0m';
+			console.error(setTextRed, `Error in Value: ${error}`);
+			if (strictBuild) {
+				throw error;
+			}
+		}
+	}
+
+	// Query data integration
```

Comment:
What's this comment for?

Reply:
that should not be there, will remove

Reply:
[af2d4db](https://github.com/evidence-dev/evidence/pull/2672/commits/af2d4dbf110bf709270453da950befd25bc2f5c2)